docs/details/vision.dox

```diff
@@ -133,20 +133,70 @@ equal to the number of features contained in the query array.
 \defgroup cv_func_nearest_neighbour nearestNeighbour
 \ingroup featmatcher_mat
 
-\brief Nearest Neighbour
-
-Calculates nearest distances between two 2-dimensional arrays containing
-features based on the type of distance computation chosen. Currently, \ref
-AF_SAD (sum of absolute differences), \ref AF_SSD (sum of squared differences)
-and \ref AF_SHD (hamming distance) are supported.
-One of the arrays containing the training data and the other the
-query data. One of the dimensions of the both arrays must be equal among them,
-identifying the length of each feature. The other dimension indicates the
-total number of features in each of the training and query arrays. Two
-1-dimensional arrays are created as results, one containg the smallest N
-distances of the query array and another containing the indices of these
-distances in the training array. The resulting 1-dimensional arrays have length
-equal to the number of features contained in the query array.
+\brief Determine the nearest neighbouring points to a given set of points
+
+In this function, a "point" is simply a geometric point's coordinates in an
+n-dimensional space. The coordinates are described along dim0 (much like a
+traditional column vector). The \p train and \p query arrays are each expected
+to be a list of such coordinates, enumerated along dim1 (however, the usage of
+dim0 and dim1 here can also be switched, as explained later).
+
+This function calculates which points in the \p train array are nearest to each
+point in the \p query array, based on the distance metric specified by \p
+dist_type: \ref AF_SAD (sum of absolute differences), \ref AF_SSD (sum of
+squared differences, the default option), or \ref AF_SHD (hamming distance). The
+resulting \p n_dist nearest neighboring points are described in two output
+arrays:
+- \p idx:  contains the index of each result that corresponds to the point in
+           \p train
+- \p dist: contains the distance from the query point to the result's
+           corresponding point in \p train
+
+In both the output arrays \p idx and \p dist, the set of nearest neighbor
+results are enumerated along dim0, in which the ith result is the ith nearest
+point to the query point. The result sets for the queries are placed in
+consecutive columns along dim1, in the same order as the queries are listed
+along dim1 of \p query. The output arrays therefore will have a shape of \p
```

Comment:
If we put therefore in the middle, we need to put lot of punctuations like `... arrays;  therefore, ...` but that doesn't make sense either because the sentence prior to `therefore` is not complete even partially.

`Therefore, the output arrays ...` is correct.